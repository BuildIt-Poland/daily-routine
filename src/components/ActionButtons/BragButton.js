import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { role } from '../../types';
import { BRAG_ROLE_ACTION } from '../../constants/roles';
import { colorGreen, colorDarkGreen } from '../../styles/designTokens';
import LightbulbIcon from './LightbulbIcon';
import ButtonBase from './ButtonBase';

const StyledButton = styled(ButtonBase)`
  background-color: ${colorGreen};
  border-bottom-color: ${colorDarkGreen};
`;

const BUTTON_LABEL = 'Brag about my efforts';
const UPDATED_BUTTON_LABEL = 'Brag more';

function BragButton({ quoteID, role, ...props }) {
  const [text, setText] = useState(BUTTON_LABEL);

  const updateButtonLabel = () => setText(UPDATED_BUTTON_LABEL);

  return (
    <StyledButton
      {...props}
      to={`/${role}/${BRAG_ROLE_ACTION}/${quoteID}`}
      icon={LightbulbIcon}
      onClick={updateButtonLabel}
    >
      {text}
    </StyledButton>
  );
}

BragButton.propTypes = {
  role: role.isRequired,
  quoteID: PropTypes.string.isRequired
};

export default BragButton;
