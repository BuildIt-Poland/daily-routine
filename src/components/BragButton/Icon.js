import React from 'react';
import styled from 'styled-components';

import { Lightbulb } from '../Icons';

const Wrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: -1rem;
`;

function Icon() {
  return (
    <Wrapper>
      <Lightbulb />
    </Wrapper>
  );
}

export default Icon;
