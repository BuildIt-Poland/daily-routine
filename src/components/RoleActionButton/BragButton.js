import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { colorGreen, colorDarkGreen } from '../../styles/designTokens';
import Button from '../Button';
import LightbulbIcon from './LightbulbIcon';

const StyledButton = styled(Button)`
  background-color: ${colorGreen};
  border-bottom-color: ${colorDarkGreen};
`;

function BragButton({ to }) {
  return (
    <StyledButton as={Link} to={to}>
      Brag about my efforts
      <LightbulbIcon />
    </StyledButton>
  );
}

BragButton.propTypes = {
  to: PropTypes.string.isRequired
};

export default BragButton;
