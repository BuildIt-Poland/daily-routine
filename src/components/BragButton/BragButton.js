import React from 'react';

import { BRAG } from '../../constants/buttonVariants';
import Button from '../Button';
import Icon from './Icon';

function BragButton() {
  return (
    <Button variant={BRAG} to="/" icon={<Icon />}>
      Brag about my efforts
    </Button>
  );
}

export default BragButton;
