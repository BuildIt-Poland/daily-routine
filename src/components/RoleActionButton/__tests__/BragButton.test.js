import React from 'react';
import { create } from 'react-test-renderer';

import BragButton from '../BragButton';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));
jest.mock('../LightbulbIcon', () => 'LightbulbIcon');

describe('COMPONENT - RoleButton BragButton', () => {
  it('renders correctly', () => {
    const component = create(<BragButton to="/devops/brag" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
