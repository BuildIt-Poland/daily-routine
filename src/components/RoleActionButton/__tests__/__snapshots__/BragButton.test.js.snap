// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RoleButton BragButton renders correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #21c230;
  color: #ffffff;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #138a1e;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  width: 100%;
  height: 4.7rem;
  background-color: #21c230;
  border-bottom-color: #138a1e;
}

.c0:not(:disabled) {
  cursor: pointer;
}

<Link
  className="c0"
  to="/devops/brag"
>
  Brag about my efforts
  <LightbulbIcon />
</Link>
`;
